src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -381,22 +381,58 @@ public void AddUserDefinitions(string script, CultureInfo parseCulture = null, A
             var options = new ParserOptions()
             {
                 AllowsSideEffects = false,
+                AllowParseAsTypeLiteral = true,
                 Culture = parseCulture ?? CultureInfo.InvariantCulture
             };
 
             var sb = new StringBuilder();
 
-            var parseResult = UserDefinitions.Parse(script, options);
+            var checkResult = new DefinitionsCheckResult()
+                                    .SetText(script, options);
+
+            var parseResult = checkResult.ApplyParse();
+
+            if (parseResult.HasErrors)
+            {
+                sb.AppendLine("Something went wrong when parsing user definitions.");

```

Comment:
>Something went wrong when parsing user definitions. [](http://example.com/codeflow?start=31&length=51)

@MikeStall when are we going to have localized resources for MS.PFx.Interpreter?